allennlp_rc/dataset_readers/transformer_squad.py

```diff
@@ -138,18 +138,30 @@ def make_instances(
         first_answer_offset: Optional[int],
     ) -> Iterable[Instance]:
         # tokenize context by spaces first, and then with the wordpiece tokenizer
-        # For RoBERTa, this produces a bug where every token is marked as beginning-of-sentence. The huggingface
-        # implementation has the same bug, so we're leaving it in.
+        # For RoBERTa, this produces a bug where every token is marked as beginning-of-sentence. To fix it, we
```

Comment:
Presumably this will be fixed in huggingface at some point? Can you leave a TODO with a reference to the bug?

Reply:
It's not really a bug in huggingface. It's a problem with tokenizing twice. RoBERTa tokenizers mark words that are preceded by a space in a certain way. If you tokenize by space first, and then re-tokenize the fragments, you lose the space, and the RoBERTa tokenizer switches behavior.